### ISSUE TITLE
When changing terminal size terminal scroll-back isn't cleared.

### ISSUE THREAD

**Issue by GoodClover** [makeworld-the-better-one/amfora#183]
For a demonstration of the bug, a video explains best:
![Amfora scroll-back bug demonstration video](https://user-images.githubusercontent.com/27222523/107122094-9f773800-688d-11eb-88f3-e80364a2ede9.mp4)
(This is Windows Terminal with cmd)

When you either change your terminal's font size (what I did in the video) or resize the terminal window, Amfora re-renders everything to fit (already better than some programs :).
However, it appears that rather than clearing the entire screen + scroll-back it just clear the screen and leaves a messed up scroll-back.

I presume that this caused by a use of the ED escape sequence mode 2 instead of mode 3, meaning that the screen is cleared and not the scroll-back.
Either that or a different sequence of escape codes that is not ED is being used to clear the screen, in that case it should be switched to ED.

![image](https://user-images.githubusercontent.com/27222523/107122213-4a87f180-688e-11eb-854d-df1f1eab5a14.png)
[Above screenshot from Wikipedia](https://en.wikipedia.org/wiki/ANSI_escape_code#CSI_(Control_Sequence_Introducer)_sequences)

*This may actually be an upstream issue, but I'm not a Go programmer so idk*

**Comment by makeworld-the-better-one** [comment]
Thanks for reporting! This is likely an upstream bug with cview or tcell. In the `cview-update` branch I have updated both those dependencies. Could you unzip and run this binary to see if the issue still occurs? The binary is for 64-bit Windows. If you're uncomfortable running the binary you can build it yourself by following the instructions on the README, but running `git checkout cview-update` before building.

[amfora.exe.zip](https://github.com/makeworld-the-better-one/amfora/files/5940127/amfora.exe.zip)

**Comment by GoodClover** [comment]
The issue appears to still be there in the executable you attached.

**Comment by makeworld-the-better-one** [comment]
I have opened https://github.com/gdamore/tcell/issues/428 to ask about this. As this is an upstream issue, this will remain open and unfixed until I understand the issue better, and likely until someone else fixes it. Thanks again for spotting this.

**Comment by gdamore** [comment]
tcell definitely does not use ED.  tcell attempts to rely on only a minimal set of terminal features unless they bring particular value.  Our redraw doesn't clear the screen at all, instead we redraw the entire screen.

I will look into supporting this escape code, however adding support for new escape codes is kind of a pain because we never know what the various terminals (and terminfo descriptions) have.

**Comment by makeworld-the-better-one** [comment]
@gdamore Thanks for responding, I understand the value in sticking to simple features.

Do you have suggestions for what cview could do to mitigate this while waiting for the possible fix?

**Comment by makeworld-the-better-one** [comment]
I opened [this](https://gitlab.com/tslocum/cview/-/issues/56) cview issue] to notify @tslocum about this.

@gdamore Any ideas about how to fix this?

**Comment by gdamore** [comment]
I will eventually try to get this done in tcell.  Unfortunately I haven't had much time to work on tcell (it doesn't really pay any bills), but hopefully I'll get to it soon.

If someone wanted to submit a PR to tcell I'd be open.

**Comment by makeworld-the-better-one** [comment]
@GoodClover could you confirm something? Does scrolling with the mouse work normally? Like without font size changes or window resizes like you mention. When you scroll with the mouse on a Gemini page, does it work?

**Comment by GoodClover** [comment]
Nope, you can't scroll normally, only when the window is resized/font changed/etc.

**Comment by makeworld-the-better-one** [comment]
Ok, I believe I understand the issue then, as I explain [here](https://gitlab.com/tslocum/cview/-/issues/56#note_539435991). I will work with the cview dev towards a fix. The actual issue is that scrolling is broken in Windows Terminal. The thing shown in your video is just a side effect.

**Comment by GoodClover** [comment]
Oh, sorry about the fuss them.
If it's an issue with [Windows Terminal](https://github.com/microsoft/terminal) then it's probably best to just get them to fix it rather than make a workaround?

**Comment by makeworld-the-better-one** [comment]
I'm not entirely sure if what they're doing can be labelled as a bug. If I can fix it on my end I will.

**Comment by makew0rld** [comment]
If tcell is updated in Amfora this may be fixed, see [this comment](https://github.com/gdamore/tcell/issues/428#issuecomment-1100795582).
